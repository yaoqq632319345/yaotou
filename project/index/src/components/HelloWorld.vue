<template>
  <h1>{{ msg }}</h1>
  <button @click="count++">{{ count }}</button>
  <hello msg="1234"></hello>
</template>

<script lang="ts">
import { defineComponent, ref } from 'vue';
import Hello from './Hello.vue';
export default defineComponent({
  name: 'HelloWorld',
  components: {
    Hello,
  },
  props: {
    msg: String,
  },
  setup() {
    const count = ref(1);
    return {
      count,
    };
  },
});
</script>
