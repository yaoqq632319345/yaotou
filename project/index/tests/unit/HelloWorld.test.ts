import { shallowMount, mount } from '@vue/test-utils';
import HelloWorld from '@/components/HelloWorld.vue';
import Hello from '@/components/Hello.vue';
import TemplateList from '@/components/TemplateList.vue';

describe('HelloWorld.vue', () => {
  it('renders props.msg when passed', () => {
    const msg = 'new message';
    const shallowWrapper = shallowMount(HelloWorld, {
      props: { msg },
    });

    expect(shallowWrapper.html()).toMatchSnapshot();
    expect(shallowWrapper.props().msg).toBe(msg);
    expect(shallowWrapper.findAllComponents(Hello)).toHaveLength(1);
    expect(shallowWrapper.findAllComponents(TemplateList)).toHaveLength(0);

    const wrapper = mount(HelloWorld, {
      props: { msg },
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.props().msg).toBe(msg);
    expect(wrapper.findAllComponents(Hello)).toHaveLength(1);
  });
});
