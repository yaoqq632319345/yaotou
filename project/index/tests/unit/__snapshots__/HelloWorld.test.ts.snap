// Vitest Snapshot v1

exports[`HelloWorld.vue > renders props.msg when passed 1`] = `
"<h1>new message</h1>
<button>1</button>
<hello-stub msg=\\"1234\\"></hello-stub>"
`;

exports[`HelloWorld.vue > renders props.msg when passed 2`] = `
"<h1>new message</h1>
<button>1</button>
<h1 class=\\"hello\\">1234</h1>"
`;
